@vates/compose/index.test.js

```diff
@@ -9,60 +11,67 @@ const mul3 = x => x * 3
 
 describe('compose()', () => {
   it('throws when no functions is passed', () => {
-    expect(() => compose()).toThrow(TypeError)
-    expect(() => compose([])).toThrow(TypeError)
+    assert.throws(() => compose(), TypeError)
+    assert.throws(() => compose([]), TypeError)
   })
 
   it('applies from left to right', () => {
-    expect(compose(add2, mul3)(5)).toBe(21)
+    assert.strictEqual(compose(add2, mul3)(5), 21)
   })
 
   it('accepts functions in an array', () => {
-    expect(compose([add2, mul3])(5)).toBe(21)
+    assert.strictEqual(compose([add2, mul3])(5), 21)
   })
 
   it('can apply from right to left', () => {
-    expect(compose({ right: true }, add2, mul3)(5)).toBe(17)
+    assert.strictEqual(compose({ right: true }, add2, mul3)(5), 17)
   })
 
   it('accepts options with functions in an array', () => {
-    expect(compose({ right: true }, [add2, mul3])(5)).toBe(17)
+    assert.strictEqual(compose({ right: true }, [add2, mul3])(5), 17)
   })
 
   it('can compose async functions', async () => {
-    expect(
+    assert.strictEqual(
       await compose(
         { async: true },
         async x => x + 2,
         async x => x * 3
-      )(5)
-    ).toBe(21)
+      )(5),
+      21
+    )
   })
 
   it('forwards all args to first function', () => {
-    expect.assertions(1)
+    const tracker = new assert.CallTracker()
+    const callsfunc = tracker.calls(assert.deepEqual, 1)
 
     const expectedArgs = [Math.random(), Math.random()]
     compose(
       (...args) => {
-        expect(args).toEqual(expectedArgs)
+        callsfunc(args, expectedArgs)
       },
       // add a second function to avoid the one function special case
       Function.prototype
     )(...expectedArgs)
+
+    tracker.verify()
```

Comment:
I find this a bit clearer:

```js
    const tracker = new assert.CallTracker()

    const expectedArgs = [Math.random(), Math.random()]
    compose(
      tracker.calls((...args) => {
        assert.deepEqual(args, expectedArgs)
      }, 1),
      // add a second function to avoid the one function special case
      Function.prototype
    )(...expectedArgs)

    tracker.verify()
```

What do you think?

Also, I'm not sure we really need to test that the function has been called, this behavior has already been tested above :slightly_smiling_face:

Reply:
You're right, it's clearer.

I was trying to do something that could be more or less similar to the expression "expect.assertions(1)" with the assert module 🤔

Reply:
Well actually, when I saw that it would be quite complicated to implement, I thought that it would be I good idea to get rid of it but I didn't dare to do it.

Reply:
I'll do it.